### PR TITLE
chore: add basic styles and BootstrapVue components to styleguide

### PR DESCRIPTION
**Pull request by LeoniePeters**
- Adds a dummy data object with an array of thumbnail URLs, to be used as a global mixin for styleguide components
- Adds portal style customised Boostrap Vue components to the styleguide
- Adds typography, colours and border general style definitions to the styleguide
- Started a bit of clean up on existing components. [TBC](https://europeana.atlassian.net/browse/EC-6596)

### CODE REVIEW

File: packages/portal/docs/style/BootstrapVue/Alert.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Default variant
+```jsx
+  <b-alert show dismissible class="mb-4">
+    This is a default info variant alert and it is configured to be dismissible
+  </b-alert>
+  <b-alert
+    show
+    variant="dark"
+  >
+    This is a dark variant alert and it is configured to be dismissible
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Alert.md:19]: **Not** dismissible, no?

File: packages/portal/docs/style/Typography.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+More info about font styles and examples where they are applied can be found on the [Typography page on Figma](https://www.figma.com/file/TQHeWavFSkWCWGMntlthcJ/Component-Library?type=design&node-id=2330%3A6894&mode=design&t=7ZfmXiyHHKs1NQqO-1) and the [Font styles page in Confluence](https://europeana.atlassian.net/wiki/spaces/EC/pages/2279047169/Font+styles)
```
Comment (rwd) [inline packages/portal/docs/style/Typography.md:1]: Do you think it's worth moving the content of the font styles pages from Confluence to here? That way we don't have this information spread about in so many different places, especially as neither the Figma nor Confluence pages are publicly accessible.

File: packages/portal/docs/style/BootstrapVue/Alert.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+Customised BootstrapVue alert component.
+
+Docs including more features and variants: https://bootstrap-vue.org/docs/components/alert
+
+Used by:
+- AlertMessage
+- InfoMessage
+- SearchFilters (Newspaper/WWI api filter alert)
+
+Default variant
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Alert.md:10]: Only one of them is default. Either remove this line, or have two code blocks, the second with its own "dark variant" heading.

File: packages/portal/docs/style/BootstrapVue/Card.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
+Customised BootstrapVue card component. Currently it is not directly used as in the examples, but could be and forms the base of other card components.
+
+Docs: https://bootstrap-vue.org/docs/components/card
+
+Used by:
+- ContentCard
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Card.md:6]: Link to this component in the styleguide.

File: packages/portal/docs/style/BootstrapVue/Card.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+- [EntityHeader](/#/Components/Entity?id=entityheader)
+- [InfoCard](/#/Components/Generic?id=infocard)
+- [ItemPreviewCard](/#/Components/Item?id=itempreviewcard)
+- MetadataBox
+- RelatedCollectionsCard
+- [RelatedEditorial](/#/Components/Related?id=relatededitorial)
+- RelatedGalleries
+- UserCreateSetButton
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Card.md:14]: Link to this component in the styleguide.

File: packages/portal/docs/style/BootstrapVue/Modal.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+- ErrorModal
+- ItemLikeButton (limit reached)
+- ItemPinButton (limit reached)
+- ItemPinModal
+- [ItemTranscribeButton](/#/Components/Item?id=itemtranscribebutton)
+- SetAddItemModal
+- SetDeleteModal
+- SetFormModal
+- [SetPublicationRequestWidget](/#/Components/Set?id=setpublishbutton)
+- ShareSocialModal ()
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Modal.md:21]: Are the parentheses deliberate?

File: packages/portal/docs/style/BootstrapVue/Modal.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+default modal
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Modal.md:23]: Remove this line.

File: packages/portal/docs/style/BootstrapVue/Modal.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+```jsx
+<div>
+  <b-button v-b-modal.modal-1 class="mb-3 mr-3">Show default modal</b-button>
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Modal.md:26]: The buttons in this modal look misplaced. Is this expected?

File: packages/portal/docs/style/BootstrapVue/Toast.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+      Show toast
+    </b-button>
+    <b-button class="mb-2" variant="primary" @click="$bvToast.show('example-toast-2')">
+      Show white toast with buttons
+    </b-button>
+
+    <b-toast 
+      id="example-toast"
+      static
+      noCloseButton
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Toast.md:28]: The props on this should be in kebab-case.

File: packages/portal/docs/style/BootstrapVue/Toast.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+      solid
+      autoHideDelay="5000"
+      toastClass="brand-toast"
+    >
+      This toast message hides after 5s
+    </b-toast>
+    <b-toast 
+      id="example-toast-2"
+      static
+      noCloseButton
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Toast.md:38]: Kebab-case this.

File: packages/portal/docs/style/BootstrapVue/Tooltip.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
+Customised BootstrapVue tooltip component.
+
+Docs including more features: https://bootstrap-vue.org/docs/components/tooltip
+
+Used by:
```
Comment (rwd) [inline packages/portal/docs/style/BootstrapVue/Tooltip.md:5]: Order these alphabetically, and link to them where also present in the style guide.

**Review by rwd — CHANGES_REQUESTED** [review]

File: packages/portal/docs/style/BootstrapVue/Alert.md (hunk 1 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Alert.md]: Indeed. Updated to read 'not be dismissable'.

File: packages/portal/docs/style/Typography.md (hunk 2 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/Typography.md]: Good idea. I have updated to the typography table to include the info from the Confluence page. Each style now has a details row (togglable display) with more info on responsive styles and example use cases.

File: packages/portal/docs/style/BootstrapVue/Alert.md (hunk 3 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Alert.md]: Removed the line.

File: packages/portal/docs/style/BootstrapVue/Card.md (hunk 4 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Card.md]: Done.

File: packages/portal/docs/style/BootstrapVue/Card.md (hunk 5 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Card.md]: Done.

File: packages/portal/docs/style/BootstrapVue/Modal.md (hunk 6 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Modal.md]: Nope. Removed.

File: packages/portal/docs/style/BootstrapVue/Modal.md (hunk 7 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Modal.md]: Done.

File: packages/portal/docs/style/BootstrapVue/Modal.md (hunk 8 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Modal.md:25]: Yes. We always use a custom modal footer. I propose to see if we can refactor it so the default footer becomes readily usable. There is a subtask for this and other improvements to some of the base components: https://europeana.atlassian.net/browse/EC-6598

File: packages/portal/docs/style/BootstrapVue/Toast.md (hunk 9 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Toast.md]: Done.

File: packages/portal/docs/style/BootstrapVue/Toast.md (hunk 10 above)
Comment (LeoniePeters) [reply packages/portal/docs/style/BootstrapVue/Toast.md]: Done.

**Pull request merged**